packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/BillingClientFactoryImpl.java

```diff
@@ -15,11 +18,31 @@ final class BillingClientFactoryImpl implements BillingClientFactory {
 
   @Override
   public BillingClient createBillingClient(
-      @NonNull Context context, @NonNull MethodChannel channel, int billingChoiceMode) {
+      @NonNull Context context,
+      @NonNull MethodChannel channel,
+      int billingChoiceMode,
+      @Nullable UserChoiceBillingListener userChoiceBillingListener) {
     BillingClient.Builder builder = BillingClient.newBuilder(context).enablePendingPurchases();
-    if (billingChoiceMode == BillingChoiceMode.ALTERNATIVE_BILLING_ONLY) {
-      // https://developer.android.com/google/play/billing/alternative/alternative-billing-without-user-choice-in-app
-      builder.enableAlternativeBillingOnly();
+    switch (billingChoiceMode) {
+      case BillingChoiceMode.ALTERNATIVE_BILLING_ONLY:
+        // https://developer.android.com/google/play/billing/alternative/alternative-billing-without-user-choice-in-app
+        builder.enableAlternativeBillingOnly();
+        break;
+      case BillingChoiceMode.USER_CHOICE_BILLING:
```

Comment:
Should this have a docs link too, like the above?

Reply:
Done